ScriptureRenderingPipeline/Rendering.cs

```diff
@@ -26,68 +28,202 @@ public static class Rendering
+                }
+
+                if (fileType == "docx")
+                {
+                    DocxConfig config = CreateDocxConfig(req.Query);
+                    DocxRenderer renderer = new DocxRenderer(config);
+                    var output = renderer.Render(document);
+                    string outputFilePath = Path.Join(repoDir, "output.docx");
+                    using (var stream = new FileStream(outputFilePath, FileMode.Create))
+                    {
+                        output.Write(stream);
+                    }
+                    var outputStream = File.OpenRead(outputFilePath);
+                    return new FileStreamResult(outputStream, "application/octet-stream")
+                    {
+                        FileDownloadName = fileName ?? "output.docx",
+                    };
+                }
+                
+                if (fileType == "usfm")
+                {
+                    string tempFolder = CreateTempFolder();
+                    string tempZipPath = Path.Join(repoDir, "output.zip");
+                    if (isBTTWriter)
+                    {
+                        USFMRenderer renderer = new USFMRenderer();
+                        var output = renderer.Render(document);
+                        var idMarkers = document.GetChildMarkers<IDMarker>();
+                        string usfmFileName = idMarkers.Count == 0 ? "document.usfm" : idMarkers[0].TextIdentifier + ".usfm";
+                        File.WriteAllText(Path.Join(tempFolder, usfmFileName), output);
+                    }
+                    else
+                    {
+                        foreach (var file in Directory.GetFiles(repoDir, "*.*", SearchOption.AllDirectories))
                         {
-                            throw new Exception($"Error parsing {file}", ex);
+                            if (validExensions.Contains(Path.GetExtension(file)))
+                            {
+                                File.Copy(file, Path.Join(tempFolder, Path.GetFileName(file)));
+                            }
                         }
                     }
+                    ZipFile.CreateFromDirectory(tempFolder, tempZipPath);
+                    var outputStream = File.OpenRead(tempZipPath);
+                    return new FileStreamResult(outputStream, "application/octet-stream")
+                    {
+                        FileDownloadName = fileName ?? "output.zip",
+                    };
                 }
-            }
 
-            var output = renderer.Render(document);
-            string outputFilePath = Path.Join(repoDir, "output.docx");
-            using (var stream = new FileStream(outputFilePath, FileMode.Create))
-            {
-                output.Write(stream);
+                if (fileType == "pdf")
+                {
+                    var latexConverterUrl = Environment.GetEnvironmentVariable("PDFConversionEndpoint");

```

Comment:
Should this be named pdfConverterUrl?

Reply:
Possibly but I think we'll just backlog it as it is existing code already